bindings_ffi/src/mls.rs

```diff
@@ -713,32 +727,32 @@ mod tests {
 
     #[derive(Clone)]
     struct RustStreamCallback {
-        num_messages: Arc<Mutex<u32>>,
+        num_messages: Arc<AtomicU32>,
     }
 
     impl RustStreamCallback {
         pub fn new() -> Self {
             Self {
-                num_messages: Arc::new(Mutex::new(0)),
+                num_messages: Arc::new(AtomicU32::new(0)),
```

Comment:
I switched out the Mutex here b/c `AtomicU32` makes more sense IMO but also because the `stream_all_messages` test kept erroring and I was troubleshooting the problem. I think it turned out the test was not waiting long enough for the message to be received, so increasing the timeout fixed it. Although maybe @richardhuaaa knows more in that area.

it might be good to try and use `std::sync::Barrier` here or the async variant on crates.io `async-barrier` instead of tokio::sleep, but might be tricky with the callback + async